File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        index1 = 0
        index2 = len(numbers) - 1

        while index1 < index2:
            test = numbers[index1] + numbers[index2]

            if test == target:
                break
            elif test < target:
                index1+= 1
            elif test > target:
                index2-= 1

        return [index1 + 1, index2 + 1]

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }

        System.out.println(map);

        for (int i = 0; i < nums.length; i++) {
            int find = target - nums[i];

            if (map.containsKey(find) && map.get(find) != i) {
                int index1 = i,
                index2 = map.get(find);

                return new int[]{index1, index2};
            }
        }

        return new int[0];
    }
}
